packages/flutter_tools/lib/src/version.dart

```diff
@@ -125,7 +124,7 @@ class FlutterVersion {
         );
         _channel = channel.substring(slash + 1);
       } else if (channel.isEmpty) {
-        _channel = 'unknown';
+        _channel = 'local';
```

Comment:
I'll need to double check with some folks on monday that this isn't going to break anything.

Reply:
Any conclusion

Reply:
I'm sure this would break things in the release process (maybe even post-submit in the devicelab?)

Also, this branch is only reached if the output of `git rev-parse --abbrev-ref --symbolic ${u}` is empty (if they checked out a local branch, they'd reach the `else` block below. The most common reason they would have that output is a detached HEAD, which could result from `flutter version v1.10.12`. In this case, they are not on a "local" channel, they are not on any channel at all, so I think `unknown` is more correct.

For this kind of a breaking change, I would suggest opening a request for comment proposal (as an issue) first before opening a PR.